controllers/nstemplatetier/nstemplatetier_controller_test.go

```diff
@@ -77,6 +82,247 @@ func TestReconcile(t *testing.T) {
+				})
+				t.Run("don't add revisions when they are up to date", func(t *testing.T) {
+					// given
+					// the NSTemplateTier already has the revisions from previous test
+
+					// when
+					res, err = r.Reconcile(context.TODO(), req)
+					// then
+					require.NoError(t, err)
+					require.Equal(t, reconcile.Result{Requeue: false}, res) // no reconcile
+					// revisions are the same
+					tiertest.AssertThatNSTemplateTier(t, "base1ns", cl).
+						HasStatusTierTemplateRevisions(tierTemplatesRefs)
+					// expected TierTemplateRevision CRs are still there
+					ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+					err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(base1nsTier.GetNamespace()))
+					require.NoError(t, err)
+					require.Len(t, ttrs.Items, len(tierTemplatesRefs)) // it's one TTR per each tiertemplate in the NSTemplateTier
+					t.Run("revision field is set but some TierTemplateRevision is missing, status should be updated", func(t *testing.T) {
+						// given
+						// the NSTemplateTier already has the revisions from previous test
+						// we delete the first TTR to make sure the status is updated and the TTR get's recreated
+						err = cl.Delete(context.TODO(), &ttrs.Items[0])
+						require.NoError(t, err)
+
+						// when
+						res, err = r.Reconcile(context.TODO(), req)
+						// then
+						require.NoError(t, err)
+						// revisions are the same
+						tiertest.AssertThatNSTemplateTier(t, "base1ns", cl).
+							HasStatusTierTemplateRevisions(tierTemplatesRefs)
+						// expected TierTemplateRevision CRs are there
+						ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+						err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(base1nsTier.GetNamespace()))
+						require.NoError(t, err)
+						require.Len(t, ttrs.Items, len(tierTemplatesRefs)) // it's one TTR per each tiertemplate in the NSTemplateTier
+					})
+				})
+
+			})
+
+			t.Run("revision field is set but TierTemplateRevision CRs are missing, they should be created", func(t *testing.T) {
+				// given
+				// the NSTemplateTier has already the status.revisions field populated
+				// but the TierTemplateRevision CRs are missing
+				tierTemplates := initTierTemplates(t, withTemplateObjects(crq), base1nsTier.Name)
+				base1nsTierWithRevisions := base1nsTier
+				initialRevisions := map[string]string{
+					"base1ns-admin-123456new":            "base1ns-admin-123456new-abcd",
+					"base1ns-clusterresources-123456new": "base1ns-clusterresources-123456new-abcd",
+					"base1ns-code-123456new":             "base1ns-code-123456new-abcd",
+					"base1ns-dev-123456new":              "base1ns-dev-123456new-abcd",
+					"base1ns-edit-123456new":             "`base1ns-edit-123456new-abcd",
+					"base1ns-stage-123456new":            "base1ns-stage-123456new-abcd",
+					"base1ns-viewer-123456new":           "base1ns-viewer-123456new-abcd",
+				}
+				base1nsTierWithRevisions.Status.Revisions = initialRevisions
```

Comment:
`base1nsTier` is a pointer so this will also overwrite the `.Status.Revisions` of `base1nsTier` which is used in all other subtests. Is this intended?

Reply:
No it was not intentional , thanks for pointing this out. I've created a new pointer in https://github.com/codeready-toolchain/host-operator/pull/1103/commits/e00a40e2bb8460661654dfddbbd71253677828fe